src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -157,17 +159,23 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
     formFieldHandlers,
   );
 
+  useEffect(() => {
+    if (tempInitialValues) {
+      setInitValues(tempInitialValues);
+    }
+  }, [tempInitialValues]);
+
```

Comment:
Why do you need this?

Reply:
Did you mean the useEffect?

Reply:
Yes, I'm asking because when introducing new rendering types most of the time we don't have reason to touch this level of code as it's already abstracted at a high level, that's assuming the standard patterns are followed.

Reply:
Once the initial values are loaded from useInitialValues hook, I am passing down the initial values to the field via context, hence this snippet which updates the state variable holding the initial values.

Reply:
Again, you don't need this change if you properly follow the standard patterns. Kindly consider introducing a handler.

Reply:
Will do, thank you.

Reply:
Could we also have a session to discuss this further?